proxy/pkg/zdmproxy/querymodifier.go

```diff
@@ -163,3 +154,41 @@ func queryOrPrepareRequiresQueryReplacement(statementsQueryData []*statementQuer
 
 	return requiresQueryReplacement(statementsQueryData[0]), statementsQueryData[0], nil
 }
+
+func (recv *QueryModifier) assignRequestId(protoVer primitive.ProtocolVersion, decodedFrame *frame.Frame) (bool, error) {
+	if protoVer < primitive.ProtocolVersion4 {
+		return false, nil
+	}
+	op := decodedFrame.Header.OpCode
+	if op != primitive.OpCodePrepare && op != primitive.OpCodeExecute && op != primitive.OpCodeQuery && op != primitive.OpCodeBatch {
+		return false, nil
+	}
+
+	customPayload := decodedFrame.Body.CustomPayload
+	allow := recv.rateLimiters.Allow(RequestIdTracingLimit)
+	if !allow {
+		// remove ID potentially provided by the upstream application
```

Comment:
We only enter this if tracing is enabled in the configuration so if it is not enabled the app provided request ID will remain on the frame and `.GetRequestId()` will still return a valid request id which means the request id will still be logged.

Reply:
Fixed.